File: App.client/projects/proctor-app/scripts/models/meeting.ts
import { User, MeetingStatus } from './user';
import { Complaint } from './complaint';
import { MeetingParticipant } from './meeting-participant';

export class Meeting {
  id: number;
  complaintId: number;
  scheduledBy: number;
  scheduledAt: string;
  durationMinutes: number;
  location: string;
  agenda?: string | null;
  status: string;
  outcome?: string | null;
  createdOn: string;
  modifiedOn: string;
  complaint?: Complaint | null;
  scheduler?: User | null;
  participants: MeetingParticipant[];

  constructor() {
    this.id = 0;
    this.complaintId = this.scheduledBy = 0;
    this.scheduledAt = new Date().toISOString();
    this.durationMinutes = 30;
    this.location = '';
    this.agenda = this.outcome = null;
    this.status = MeetingStatus.Scheduled;
    this.createdOn = this.modifiedOn = '';
    this.participants = [];
  }
}

File: App.Models/Meeting.cs
using App.Core.Models;

namespace App.Models
{
    public class Meeting : IAuditableEntity
    {
        public int Id { get; set; }
        public int ComplaintId { get; set; }
        public int ScheduledBy { get; set; }
        public DateTime ScheduledAt { get; set; }
        public int DurationMinutes { get; set; } = 30;
        public string Location { get; set; } = string.Empty;
        public string? Agenda { get; set; }
        public MeetingStatus Status { get; set; } = MeetingStatus.Scheduled;
        public string? Outcome { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual Complaint Complaint { get; set; } = null!;
        public virtual User Scheduler { get; set; } = null!;
        public virtual ICollection<MeetingParticipant> Participants { get; set; } = new List<MeetingParticipant>();
    }
}


